src/crafting_gui.cpp

```diff
@@ -150,7 +152,127 @@ static int print_items( const recipe &r, const catacurses::window &w, int ypos,
     return ypos - oldy;
 }
 
-const recipe *select_crafting_recipe( int &batch_size )
+void crafting_bills_display( const tripoint &loc )
+{
+    int term_x = TERMY > FULL_SCREEN_HEIGHT ? ( TERMY - FULL_SCREEN_HEIGHT ) / 2 : 0;
+    int term_y = TERMX > FULL_SCREEN_WIDTH ? ( TERMX - FULL_SCREEN_WIDTH ) / 2 : 0;
+
+    catacurses::window w_bills = catacurses::newwin( FULL_SCREEN_HEIGHT, FULL_SCREEN_WIDTH,
+                                 point( term_y, term_x ) );
+    const int entries_per_page = FULL_SCREEN_HEIGHT - 4;
+    size_t selection = 0;
+    input_context ctxt( "CRAFTING_BILLS" );
+    ctxt.register_cardinal();
+    ctxt.register_updown();
+    ctxt.register_action( "ANY_INPUT" );
+    ctxt.register_action( "CONFIRM" );
+    ctxt.register_action( "QUIT" );
+    ctxt.register_action( "NEW_BILL" );
+    ctxt.register_action( "REMOVE_BILL" );
+    while( true ) {
+        werase( w_bills );
+        // create a list of bills
+        crafting_bill *bill_here = g->m.crafting_bill_at( loc );
+        std::map<recipe_id, int> all_bills;
+        std::vector<recipe_id> bills_vec;
+        if( bill_here ) {
+            all_bills = bill_here->bills;
+            for( auto &elem : all_bills ) {
+                bills_vec.push_back( elem.first );
+            }
+        }
+        recipe_id cur_recipe;
+        // entries_per_page * page number
+        const size_t top_of_page = entries_per_page * ( selection / entries_per_page );
+        if( !bills_vec.empty() ) {
+            cur_recipe = bills_vec[selection];
+        }
+
+        for( int i = 0; i < FULL_SCREEN_HEIGHT - 1; i++ ) {
+            mvwputch( w_bills, point( 45, i ), BORDER_COLOR, LINE_XOXO );
+        }
+        draw_border( w_bills );
+        const nc_color col = c_white;
+        const std::string no_bills = _( "There are no bills here" );
+        if( !bills_vec.empty() ) {
+            draw_scrollbar( w_bills, selection, entries_per_page, bills_vec.size(),
+                            point( 0, 3 ) );
+            for( size_t i = top_of_page; i < bills_vec.size(); i++ ) {
+                const int y = i - top_of_page + 3;
+                trim_and_print( w_bills, point( 1, y ), 43, selection == i ? hilite( col ) : col,
+                                bills_vec[i].obj().result_name() );
+            }
+            if( selection < bills_vec.size() ) {
+                mvwprintz( w_bills, point( 46, 3 ), c_light_gray, _( "Quantity:           %d" ),
+                           all_bills[cur_recipe] );
+            } else {
+                mvwprintz( w_bills, point( 46, 4 ), c_light_red, no_bills );
+            }
+        } else {
+            mvwprintz( w_bills, point( 46, 4 ), c_light_red, no_bills );
+        }
+        mvwprintz( w_bills, point( 1, FULL_SCREEN_HEIGHT - 1 ), c_light_gray,
+                   _( "Press Enter to edit bill, N to create new bill, R to remove a bill." ) );
```

Comment:
It's better to use `input_context` to generate keybinding descriptions than hardcoding it in the text, in case the player changes the keybindings.